.github/workflows/AI.yml

```diff
@@ -50,53 +50,77 @@ jobs:
           pr_files = requests.get(diff_url, headers=headers).json()
 
           diff_text = ""
+          inline_comments = []  # Collect inline comments to post
+
           for fdata in pr_files:
               filename = fdata['filename']
               patch = fdata.get('patch', '')
               diff_text += f"File: {filename}\\nPatch:\\n{patch}\\n\\n"
 
-          # Call OpenAI for summary
+              # Call OpenAI for inline code analysis
+              issues_prompt = f"Review the following code changes for critical issues (e.g., syntax errors, undefined variables, broken imports). Provide specific inline comments for each issue with the corresponding line number from this diff:\n\n{patch}"
+              ai_headers = {"Content-Type": "application/json", "Authorization": f"Bearer {openai_key}"}
+              data_issues = {
+                  "model": "gpt-4o-mini",
+                  "messages": [{"role": "user", "content": issues_prompt}],
+                  "temperature": 0.7
+              }
+              issues_response = requests.post("https://api.openai.com/v1/chat/completions", headers=ai_headers, json=data_issues)
+              issues_response.raise_for_status()
+              issues = issues_response.json()['choices'][0]['message']['content'].strip()
+
+              # Parse issues for inline comments
+              if issues and "no issues found" not in issues.lower():
+                  for issue in issues.split("\\n- "):
+                      if issue.strip():
+                          # Example issue format: "Line X: Description of issue"
+                          if "Line " in issue:
+                              parts = issue.split(":")
+                              line_info = parts[0].strip()
+                              description = ":".join(parts[1:]).strip()
+
+                              if "Line" in line_info:
+                                  line_number = int(line_info.split(" ")[1])  # Extract line number
+                                  inline_comments.append({
+                                      "path": filename,
+                                      "line": line_number,
+                                      "side": "RIGHT",  # Changes are always on the "RIGHT" side in the diff
+                                      "body": f"**AI Code Issue Check:**\n{description}"
+                                  })
+
+          # Post inline comments as a single review
+          if inline_comments:
+              review_data = {
+                  "body": "AI-generated review comments for code issues.",
```

Comment:
The body of the review data is generic. It would be more informative to include a summary of the issues found in the comments instead of a vague statement.